### ISSUE TITLE
Building MXE in directory with special characters

### ISSUE THREAD

**Issue by starius** [mxe/mxe#1085]
Following on from #1030, MXE doesn't build in a directory with spaces in name:
```
Makefile:228: *** GNU Make chokes on paths with spaces. Stop.
```

It fails with another error if a directory contains `$`:
```
~/mxe-with-$pecial-ch@rs?-and-кириллица-!%^&()$ make -j 6
make: *** No rule to make target `/home/mxe/mxe-with-ecial-ch@rs?-and-кириллица-!%^&()/usr/i686-w64-mingw32.static/installed/a52dec', needed by `a52dec'.  Stop.
```

Can we fix it? I thought we add `'` around paths (e.g. `'$(PREFIX)/$(TARGET)/lib/pkgconfig/$(PKG).pc'`) to prevent errors like this. If we can't fix it, then error messages should be same for space and for `$`.

**Comment by tonytheodore** [comment]
Interesting. Most quoting we do is for shell commands, but we don't quote the file targets. I think we could either try quoting those targets or switching to pattern rules (the latter is something I've been meaning to try anyway).

**Comment by starius** [comment]
Commit 8aa5ccc123b429c59b453b7be2fce9ef3bea678a. Directory `/home/mxe/mxe-libmysqlclient-++-plus-plus`.

The following packages fail to build:

  * libmysqlclient (fixed in 2e71b21211db4cea44d5518cc2afda2e699361a5)
  * vtk (fixed in #1090)
  * opencv (fixed in #1093)
  * exiv2 (unrelated, fixed in 6e50094684b9f8058f9fd94af8457e4ced306b57)

Other packages were built successfully.

**Comment by tonytheodore** [comment]
Seems to be resolved.
